base/ubi9/podman-interactive-wrapper.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+set -euo pipefail
+
+# This script transparently replaces problematic 'podman run -it ...' invocations
+# with a workaround for kubedock compatibility:
+#   podman exec -it $(podman run -d --rm [OPTIONS] IMAGE tail -f /dev/null) [COMMAND]
+
+usage() {
+    echo "Usage: $0 run [OPTIONS] IMAGE [COMMAND] [ARG...]"
+    exit 1
+}
+
+ARGS=( "$@" )
+if [[ "${ARGS[0]:-}" != "run" ]]; then
+    usage
+fi
+
+RUN_OPTS=()
+IMAGE=""
+CMD_ARGS=()
+
+skip_next=0
+found_image=0
+for ((i=1; i<${#ARGS[@]}; i++)); do
+    arg="${ARGS[$i]}"
+    if [[ $skip_next -eq 1 ]]; then
+        # Only skip for --tty/--interactive with space value
+        RUN_OPTS+=("$arg")
+        skip_next=0
+        continue
+    fi
+    if [[ $found_image -eq 0 ]]; then
+        # Remove all forms of interactive/tty options for internal run
+        case "$arg" in
+            -i|-t|-it|-ti)
+                continue
+                ;;
+            --tty|--interactive)
+                # skip this and the next arg if it does not start with '-'
+                if [[ $((i+1)) -lt ${#ARGS[@]} && "${ARGS[$((i+1))]}" != "-"* ]]; then
+                    skip_next=1
+                fi
+                continue
+                ;;
+            --tty=*|--interactive=*)
+                continue
+                ;;
+            # Options that take a value
+            -u=*|--user=*)
+                # Handle -u=root and --user=root
+                RUN_OPTS+=("--user=$(id -u)")
+                continue
+                ;;
+            -u|--user)
+                # Handle -u root and --user root
+                RUN_OPTS+=("--user=$(id -u)")
+                ((i++)) # Skip the next argument (the value)
+                continue
+                ;;
+            -v|--volume|-e|--env|-w|--workdir|--name|--hostname|--entrypoint|--add-host|--device|--label|--network|--cap-add|--cap-drop|--security-opt|--tmpfs|--ulimit|--mount|--publish|--expose|--dns|--dns-search|--dns-option|--mac-address|--memory|--memory-swap|--cpu-shares|--cpus|--cpu-period|--cpu-quota|--cpu-rt-runtime|--cpu-rt-period|--cpuset-cpus|--cpuset-mems|--blkio-weight|--blkio-weight-device|--device-read-bps|--device-write-bps|--device-read-iops|--device-write-iops|--shm-size|--sysctl|--log-driver|--log-opt|--restart|--stop-signal|--stop-timeout|--health-cmd|--health-interval|--health-retries|--health-timeout|--health-start-period|--userns|--cgroup-parent|--pid|--ipc|--uts|--runtime|--storage-opt|--tmpfs|--volume-driver|--volumes-from|--env-file|--add-host|--security-opt|--device|--group-add|--init|--isolation|--kernel-memory|--label|--log-driver|--log-opt|--memory-reservation|--memory-swappiness|--oom-kill-disable|--oom-score-adj|--pids-limit|--privileged|--publish-all|--read-only|--restart|--sig-proxy|--stop-signal|--tmpfs|--ulimit|--userns|--uts|--volume|--workdir)
```

Comment:
this list has some variables listed twice e.g., `--log-opt`, `--label`, `--tmpfs`, `--device`